### ISSUE TITLE
Installing from master branch on ubuntu 14.04

### ISSUE THREAD

**Issue by vkatsikaros** [teodesian/Selenium-Remote-Driver#282]
I can't install SRD (only on ubuntu 14.04, not on 16.04) from the git repo's master branch with `dzil listdeps --missing | cpanm` and `dzil install`.

Failure in a _fresh_ 14.04

```
$ uname -a
Linux ubuntu-2gb-nyc3-01 4.4.0-38-generic #57~14.04.1-Ubuntu SMP Tue Sep 6 17:20:43 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.5 LTS
Release:    14.04
Codename:   trusty
$ perl -v | head -n 3

This is perl 5, version 18, subversion 2 (v5.18.2) built for x86_64-linux-gnu-thread-multi
(with 44 registered patches, see perl -V for more detail)

$ apt-get install git libdist-zilla-plugin-git-perl build-essential cpanminus libexpat1-dev
$ git clone https://github.com/gempesaw/Selenium-Remote-Driver
$ git config --global user.email "you@example.com"
$ git config --global user.name "Your Name"

$ cd Selenium-Remote-Driver/
$ dzil authordeps --missing | cpanm
# some packages don't get cleanly installed, force install
$ dzil authordeps --missing | cpanm --force
$ dzil listdeps --missing | cpanm
$ dzil install
[DZ] building distribution under .build/uLVQHWRBnP for installation
[DZ] beginning to build Selenium-Remote-Driver
[Git::NextVersion] Bumping version from 1.00 to 1.01
[DZ] guessing dist's main_module is lib/Selenium/Remote/Driver.pm
[TravisYML] Rebuilding .travis.yml
[DZ] writing Selenium-Remote-Driver in .build/uLVQHWRBnP
Attribute (build_root) does not pass the type constraint because: Validation failed for 'MooseX::Types::Path::Class::Dir' with value Dist::Zilla::Path=[ ".build/uLVQHWRBnP", ".build/uLVQHWRBnP", "", ".build/", "uLVQHWRBnP" ] at accessor Dist::Zilla::Plugin::Git::CommitBuild::build_root (defined at /usr/share/perl5/Dist/Zilla/Plugin/Git/CommitBuild.pm line 66) line 8.
    Dist::Zilla::Plugin::Git::CommitBuild::build_root('Dist::Zilla::Plugin::Git::CommitBuild=HASH(0x4927fd0)', 'Dist::Zilla::Path=ARRAY(0x2dc8528)') called at /usr/share/perl5/Dist/Zilla/Plugin/Git/CommitBuild.pm line 95
    Dist::Zilla::Plugin::Git::CommitBuild::after_build('Dist::Zilla::Plugin::Git::CommitBuild=HASH(0x4927fd0)', 'HASH(0xba51800)') called at /usr/local/share/perl/5.18.2/Dist/Zilla/Dist/Builder.pm line 376
    Dist::Zilla::Dist::Builder::build_in('Dist::Zilla::Dist::Builder=HASH(0x3549df0)', 'Dist::Zilla::Path=ARRAY(0x2dc8528)') called at /usr/local/share/perl/5.18.2/Dist/Zilla/Dist/Builder.pm line 421
    Dist::Zilla::Dist::Builder::ensure_built_in('Dist::Zilla::Dist::Builder=HASH(0x3549df0)', 'Dist::Zilla::Path=ARRAY(0x2dc8528)') called at /usr/local/share/perl/5.18.2/Dist/Zilla/Dist/Builder.pm line 672
    Dist::Zilla::Dist::Builder::ensure_built_in_tmpdir('Dist::Zilla::Dist::Builder=HASH(0x3549df0)') called at /usr/local/share/perl/5.18.2/Dist/Zilla/Dist/Builder.pm line 700
    Dist::Zilla::Dist::Builder::install('Dist::Zilla::Dist::Builder=HASH(0x3549df0)', 'HASH(0x2da8020)') called at /usr/local/share/perl/5.18.2/Dist/Zilla/App/Command/install.pm line 49
    Dist::Zilla::App::Command::install::execute('Dist::Zilla::App::Command::install=HASH(0x114fe38)', 'Getopt::Long::Descriptive::Opts::__OPT__::2=HASH(0x1522860)', 'ARRAY(0x1245bf8)') called at /usr/local/share/perl/5.18.2/App/Cmd.pm line 468
    App::Cmd::execute_command('Dist::Zilla::App=HASH(0x13f9390)', 'Dist::Zilla::App::Command::install=HASH(0x114fe38)', 'Getopt::Long::Descriptive::Opts::__OPT__::2=HASH(0x1522860)') called at /usr/local/share/perl/5.18.2/App/Cmd.pm line 321
    App::Cmd::run('Dist::Zilla::App') called at /usr/local/bin/dzil line 15
```

Success in a _fresh_ 16.04

```
$ uname -a
Linux ubuntu-2gb-nyc3-01 4.4.0-38-generic #57-Ubuntu SMP Tue Sep 6 15:42:33 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.1 LTS
Release:    16.04
Codename:   xenial
$ perl -v | head -n 3

This is perl 5, version 22, subversion 1 (v5.22.1) built for x86_64-linux-gnu-thread-multi
(with 58 registered patches, see perl -V for more detail)

$ apt-get install git libdist-zilla-plugin-git-perl build-essential cpanminus libexpat1-dev
$ git clone https://github.com/gempesaw/Selenium-Remote-Driver
$ git config --global user.email "you@example.com"
$ git config --global user.name "Your Name"

$ cd Selenium-Remote-Driver/
$ dzil authordeps --missing | cpanm
# some packages don't get cleanly installed, force install
$ dzil authordeps --missing | cpanm --force
$ dzil listdeps --missing | cpanm
$ dzil install
[DZ] building distribution under .build/V8VKgnd54A for installation
[DZ] beginning to build Selenium-Remote-Driver
[Git::NextVersion] Bumping version from 1.00 to 1.01
[DZ] guessing dist's main_module is lib/Selenium/Remote/Driver.pm
[TravisYML] Rebuilding .travis.yml
[DZ] writing Selenium-Remote-Driver in .build/V8VKgnd54A
[CopyFilesFromBuild] Copied .build/V8VKgnd54A/cpanfile to cpanfile
--> Working on .
Configuring Selenium-Remote-Driver-1.01 ... OK
Building and testing Selenium-Remote-Driver-1.01 ... OK
Successfully installed Selenium-Remote-Driver-1.01
1 distribution installed
[DZ] all's well; removing .build/V8VKgnd54A
```

**Comment by vkatsikaros** [comment]
Forgot to mention the repo version:
```
$ git log --oneline  -1
c71530b Merge pull request #281 from richi235/master
```
and I tried the failure on 14.04 also with
```
$ git checkout v1.00
$ git log --oneline  -1
91b7be4 Release v1.00 of Selenium::Remote::Driver
```

**Comment by teodesian** [comment]
Suspect this is due to the dzil version packaged in 14.04 -- I'll look into it though, and if so add a minimum version requirement to dist.ini.

**Comment by teodesian** [comment]
This might be fixed by some of the changes I made to dist.ini instituting TestingMania.  I'll investigate this on a 14.04 host tomorrow.

**Comment by vkatsikaros** [comment]
Thanks @teodesian! I tested on Ubuntu 14.04.5 LTS all ways of installing (since I was there :) ).

It much looks better now, but there seems an issue with sth in `Pod::Weaver`. I don't have any `Pod::Weaver` experience, but let me know if there is anyway I can help.

PS let me clarify that I originally reported the bug in case it masks another problem, or in case it helps improving sth in the module. I don't use dzilla to install the module.

Fine
--------
```
cpanm Selenium::Remote::Driver
```
and
```
git clone https://github.com/gempesaw/Selenium-Remote-Driver --branch build/master --single-branch --depth 1
cd Selenium-Remote-Driver/
perl Makefile.PL
make
make test
make install
```
Problem
-------------
However, Dist::Zilla seems to have another (better looking) issue:
```
git clone https://github.com/gempesaw/Selenium-Remote-Driver --branch build/master --single-branch --depth 1
# as expected some dev packages need installing
# Dist::Zilla::Plugin::ContributorsFromGit needs force install
dzil listdeps --missing | cpanm
dzil authordeps --missing | cpanm

dzil install
[DZ] building distribution under .build/z9H3doP2nn for installation
[DZ] beginning to build Selenium-Remote-Driver
[PkgVersion] skipping lib/Selenium/ActionChains.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/CanStartBinary.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/CanStartBinary/FindBinary.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/CanStartBinary/ProbePort.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Chrome.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Firefox.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Firefox/Binary.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Firefox/Profile.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/InternetExplorer.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/PhantomJS.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/Commands.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/Driver.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/Driver/CanSetWebdriverContext.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/Driver/Firefox/Profile.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/ErrorHandler.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/Finders.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/Mock/Commands.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/Mock/RemoteConnection.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/RemoteConnection.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/WDKeys.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Remote/WebElement.pm: assigns to $VERSION
[PkgVersion] skipping lib/Selenium/Waiter.pm: assigns to $VERSION
[PkgVersion] skipping lib/Test/Selenium/Chrome.pm: assigns to $VERSION
[PkgVersion] skipping lib/Test/Selenium/Firefox.pm: assigns to $VERSION
[PkgVersion] skipping lib/Test/Selenium/InternetExplorer.pm: assigns to $VERSION
[PkgVersion] skipping lib/Test/Selenium/PhantomJS.pm: assigns to $VERSION
[PkgVersion] skipping lib/Test/Selenium/Remote/Driver.pm: assigns to $VERSION
[PkgVersion] skipping lib/Test/Selenium/Remote/Role/DoesTesting.pm: assigns to $VERSION
[PkgVersion] skipping lib/Test/Selenium/Remote/WebElement.pm: assigns to $VERSION
[DZ] guessing dist's main_module is lib/Selenium/Remote/Driver.pm
Unknown POD in SEE ALSO: Pod::Elemental::Element::Pod5::Command at /usr/local/share/perl/5.18.2/Pod/Weaver/Section/SeeAlso.pm line 112.
```
I have
```
# cpanm Pod::Elemental::Element::Pod5::Command
Pod::Elemental::Element::Pod5::Command is up to date. (0.103004)
```

**Comment by teodesian** [comment]
So, it looks like 

dzil listdeps --missing

only is finding one dep at a time with listdeps for some reason, and then throwing an exception.  I had to run it through 10 times until it straightened up and flew right; must be a bug in the dzil version's parsing of PW deps.

Got it to build after that.  I think the long-term solution here is either to not worry about this, or to remove reliance on some of these more complicated weaver directives.

Also be aware when doing dzil test --all, the POD link check will fail -- I had to install Test::Spec WWW::Mechanize and Test::WWW::Selenium to overcome a bug with the installed perl version's POD engine.

**Comment by vkatsikaros** [comment]
It seems that this is no longer an issue. On Ubuntu 14.04:
```
$ uname -a
Linux ubuntu-s-2vcpu-4gb-lon1-01 3.13.0-149-generic #199-Ubuntu SMP Thu May 17 10:12:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
$ perl -v | head -n 3

This is perl 5, version 18, subversion 2 (v5.18.2) built for x86_64-linux-gnu-thread-multi
(with 52 registered patches, see perl -V for more detail)

$ dzil --version
dzil (Dist::Zilla::App) version 6.012 (/usr/local/bin/dzil)
```
This works ok
```
$ apt-get update
$ apt-get install git libdist-zilla-plugin-git-perl build-essential cpanminus libexpat1-dev
$ git clone https://github.com/gempesaw/Selenium-Remote-Driver
$ git config --global user.email "you@example.com"
$ git config --global user.name "Your Name"
$ git log --oneline -1
ed84528 Fix #385: Add Selenium::Edge to satisfy the need for direct connections
$ cd Selenium-Remote-Driver/
$ dzil authordeps --missing | cpanm
$ dzil listdeps --missing | cpanm
$ dzil install
$ dzil --version
$ perl -MSelenium::Firefox  -e 'print "$Selenium::Remote::Driver::VERSION\n"';
1.28
```
